src/controllers/tripRequestController.js

```diff
@@ -0,0 +1,205 @@
+/* eslint-disable no-unused-expressions */
+import { date } from 'joi';
+import models from '../database/models';
+import UserService from '../services/user.service';
+import { isBeforeToday, isBeforeDepartureDate } from '../helpers/dateValidator';
+import isNumber from '../helpers/numberValidator';
+
+const { checkManager } = UserService;
+
+const { User, TripRequest } = models;
+
+class tripController {
+  static requestAtrip = async (req, res) => {
+    const check = {
+      id: req.body.managerId,
+    };
+    const managerExists = await checkManager(check);
+    if (managerExists !== 'Found') {
+      return res.status(400).json({ message: 'Manager does not Exist' });
+    }
+    const backDated = await isBeforeToday(req.body.departureDate);
+    if (backDated !== true) {
+      return res.status(400).json({
+        message: `Departure date must not be back dated`,
```

Comment:
@Robertishimwe improve validation. ``Departure date must be a valid date`` and make sure you validate the date standard that you are using define if it is International date standard which is like ``yyyy-mm-dd``.